sql: add SHOW SURVIVAL GOAL FROM DATABASE

otan opened the pull request:
two commits in here, one is a minor fix up i thought might as well go in with this one.

----

delegate: add database name to SHOW REGIONS FROM DATABASE

This seems to match behaviour I've seen from other places, e.g. SHOW
ZONE CONFIGURATION.

Release note (sql change): The database name is now displayed in `SHOW
REGIONS FROM DATABASE`.

sql: add SHOW SURVIVAL GOAL FROM DATABASE

Release note (sql change): Add `SHOW SURVIVAL GOAL FROM DATABASE
[database]`, which shows the survival goal for a multi-region database.

otan committed:
delegate: add database name to SHOW REGIONS FROM DATABASE

This seems to match behaviour I've seen from other places, e.g. SHOW
ZONE CONFIGURATION.

Release note (sql change): The database name is now displayed in `SHOW
REGIONS FROM DATABASE`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/57278)
<!-- Reviewable:end -->

otan committed:
sql: add SHOW SURVIVAL GOAL FROM DATABASE

Release note (sql change): Add `SHOW SURVIVAL GOAL FROM DATABASE
[database]`, which shows the survival goal for a multi-region database.

ajstorm reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 2 files at r1, 11 of 11 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/57278#-:-MNV3bNaD3r9ARXb9sM-:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
thanks!

bors r=ajstorm
